python/tk_framework_desktopserver/server_protocol.py

```diff
@@ -89,15 +89,18 @@ def connectionLost(self, reason):
             # Known certificate error. These work for firefox and safari, but chrome rejected certificate
             # are currently indistinguishable from lost connection. This is true as of July 21st, 2015.
             certificate_error = False
-            certificate_error |= (
-                reason.type is OpenSSL.SSL.Error
-                and reason.value.message[0][2] == "ssl handshake failure"
-            )
-            certificate_error |= (
-                reason.type is OpenSSL.SSL.Error
-                and reason.value.message[0][2] == "tlsv1 alert unknown ca"
-            )
+            if hasattr(reason.value, "message"):
```

Comment:
Yes, I kind of agree here. We should try to retrieve "reason.value.message[0][2]" in a try/except block and then run the following lines. It will be even more fault tolerant

Reply:
Good call guys, thank you